task_2/task_2.py

```diff
@@ -0,0 +1,430 @@
+# logging.debug("Debug message")
+
+
+# 1.  Set up new file logger and don’t forget to log information about each step
+def create_logger(work_path, log_file_name, log_level):
+    # [DEBUG, INFO, WARNING, ERROR, CRITICAL]
+    print(f'Path\nwork_path = {work_path}\nlog_file_name = {log_file_name}\nlog_level = {log_level}')
+    if not os.path.exists(work_path):
+        os.makedirs(work_path)
+    path = os.path.join(work_path, log_file_name)
+    logging.basicConfig(filename=path,
+                        # level=logging.WARNING,
+                        # level=logging.DEBUG,
+                        # level=logging.INFO,
+                        # level=logging.ERROR,
+                        # level=logging.CRITICAL,
+                        level=log_level,
+                        format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+    logger = logging.getLogger("Def_Create_logger")
+    # logger.setLevel(logging.DEBUG)
+    # logger.setLevel(log_level)
+    logging.info('-------- Start script ---------')
+    logger.debug('Logger start')
+
+
+# 2.  Explore the data provider site (https://randomuser.me/documentation) and
+#     download a csv file with 5k user accounts using your script.
+def get_random_users_list(url, file_out):
+    """
+    Call the data provider site https://randomuser.me/documentation and
+    download a csv file with 5k user accounts using your script.
+    :param url:
+    :param file_out:
+    :return: users dict list
+    """
+    logger = logging.getLogger('get_user()')
+    logger.debug(f'requests url:{url}\nresult in file_out{file_out}')
+    try:
+        response = requests.get(url=url)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.error("An error occurred: %s", e)
+        print(f"An error occurred: {e}")
+    #     break
+    else:
+        # users_dict_list = []
+        count_users = response.text.count('\n')
+
+        logger.info('Request data is OK')
+        logger.info('Request url:' + response.url)
+        logger.info(f'Downloaded {count_users} accounts of users.')
+
+        with open(file_out, "w", newline='', encoding="utf-8") as file:
+            file.write(response.text)
+            logger.info(f'Users data save in file {file_out}')
+
+        # users_dict_list = [csv.DictReader(response.text.splitlines()[1:], fieldnames=response.text.splitlines()[0].split(','))]
+        # logger.debug('Create dict to users - users_dict_list')
+
+        return count_users
+
+
+# 3.  User should provide the following parameters to the command:
+#     - Destination folder -- path to a folder where output file is going to be placed (required)
+#     - Filename (default – output)
+#     - One of optional arguments:
+#         -- gender to filter the data by
+#         -- number of rows to filter by
+#     - optional positional argument -- log_level
+
+def main(output_folder, output_file_name, filter_gender, filter_rows, log_level):
```

Comment:
main must be at the top or at the bottom of the file.
if you put it in the middle it make the other programmers confused